[split from #4773] dynamo/test_misc_xpu.py::test_unbacked_sources_scalar stale test body (frame_count 0 vs 1)

### 🐛 Describe the bug

parent_issue: 4773
test_cases:
  - op_ut,third_party.torch-xpu-ops.test.xpu.dynamo.test_misc_xpu.MiscTests,test_unbacked_sources_scalar
---

## Parent Issue

Split from #4773 — [Bug Skip]: [Regression]UT failures 2026-8-2

## Affected Test Cases

- `test/xpu/dynamo/test_misc_xpu.py::MiscTests::test_unbacked_sources_scalar`

## Root Cause

`test/xpu/dynamo/test_misc_xpu.py` is a fork of upstream `pytorch/test/dynamo/test_misc.py`. The upstream test body was changed but the XPU fork still carries the old one.

XPU fork (stale):

```python
@torch.compiler.config.patch(unbacked_sources="L['x']")
def test_unbacked_sources_scalar(self):
    counter = CompileCounter()

    @torch.compile(backend=counter)
    def fn(x):
        return x * x       # <-- old body

    fn(0); fn(1); fn(2)
    self.assertEqual(counter.frame_count, 1)
```

Upstream `test/dynamo/test_misc.py:11076` (current):

```python
@torch.compiler.config.patch(unbacked_sources="L['x']")
def test_unbacked_sources_scalar(self):
    counter = CompileCounter()

    @torch.compile(backend=counter)
    def fn(x):
        return torch.ones(2) * x   # <-- new body

    fn(0); fn(1); fn(2)
    self.assertEqual(counter.frame_count, 1)
```

With the old body (`x * x`, both operands scalar), dynamo no longer produces a compiled frame at all under `unbacked_sources`, so `counter.frame_count` is `0` instead of `1`. The upstream body was rewritten to keep a tensor in the graph so the frame is still compiled. This is a stale-test-copy problem, not an XPU bug.

**Fix**: re-sync `test_unbacked_sources_scalar` in `test/xpu/dynamo/test_misc_xpu.py` with upstream `test/dynamo/test_misc.py`. 

## Reproducer

```bash
source ~/.bashrc && cd third_party/torch-xpu-ops/test/xpu && python -m pytest -sxv dynamo/test_misc_xpu.py -k "test_unbacked_sources_scalar"
```

## Representative Error

```
=================================== FAILURES ===================================
____________________ MiscTests.test_unbacked_sources_scalar ____________________
Traceback (most recent call last):
  File ".../test/xpu/dynamo/test_misc_xpu.py", line 9101, in test_unbacked_sources_scalar
    self.assertEqual(counter.frame_count, 1)
  File "/home/tszulist/src/pytorch/torch/_dynamo/test_case.py", line 127, in assertEqual
    return super().assertEqual(x, y, *args, **kwargs)
  File "/home/tszulist/src/pytorch/torch/testing/_internal/common_utils.py", line 4844, in assertEqual
    raise error_metas.pop()[0].to_error(  # type: ignore[index]
AssertionError: Scalars are not equal!

Expected 1 but got 0.
Absolute difference: 1
Relative difference: 1.0
=========================== short test summary info ============================
FAILED dynamo/test_misc_xpu.py::MiscTests::test_unbacked_sources_scalar
====================== 1 failed, 684 deselected in 6.24s =======================
```

### Versions

- Latest good PyTorch: [f2fa815](https://github.com/pytorch/pytorch/commit/f2fa81593a74bb8aca2c3da562765378665af089)
- Current (failing) PyTorch: [4ee8fea](https://github.com/pytorch/pytorch/commit/4ee8fea1f8cbb27acf0dac1c4f2e8b8bb6aef7f7)
- Platform: Intel(R) Data Center GPU Max 1100, Ubuntu 24.04, libze1 1.24.0.0, intel-opencl-icd 25.18.33578.51
- XPU build: 20260100
- Reproduced locally on `torch 2.14.0a0+gitfdd065a`, torch-xpu-ops `1e6e7fbe`.

--- a/test/xpu/dynamo/test_misc_xpu.py
+++ b/test/xpu/dynamo/test_misc_xpu.py
@@ -9092,7 +9092,7 @@ not ___dict_contains('cccccccc', G['sys'].modules)""",
 
         @torch.compile(backend=counter)
         def fn(x):
-            return x * x
+            return torch.ones(2) * x
 
         fn(0)
         fn(1)

